app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -132,6 +134,15 @@ class WireActivityViewModel @Inject constructor(
     val observeSyncFlowState: StateFlow<SyncState?> = _observeSyncFlowState
 
     init {
+        viewModelScope.launch(dispatchers.io()) {
+            when (checkSystemIntegrity()) {
+                CheckSystemIntegrityUseCase.Result.Failed -> globalAppState = globalAppState.copy(jailBreakDetected = true)
+                CheckSystemIntegrityUseCase.Result.Success -> initialize()
```

Comment:
by default  `jailBreakDetected = false`, which means app will start "drawing" a standard view, and then "re-draw" it if jailBreak is detected.
Is it okay that app draws standard view while `initialize()` is not called yet?

Reply:
I am not sure at this point :S not sure how this could be problematic if all the sessions are removed ? @vitorhugods do you see anything going wrong ?

Reply:
I guess it's a trade-off between initialization speed and total data safety I guess.

Currently, the first time the app detects that the phone is rooted:
1. Start drawing screen normally (maybe login screen, maybe conversation screen, not sure yet)
2. App detects that the phone is rooted
3. App wipes all data
4. Re-draw, but now everything is white


The second time:
1. Start drawing screen normally (will draw the Login screen, because the data was wiped)
2. App detects that the phone is rooted
3. App wipes all data
4. Re-draw, but now everything is white


Both seem fine to me. As the root detection is blazing fast anyway.

Reply:
I tested it and it worked fine everytime I am not sure if ther eis risc we get the false value before we starting to setContent 🤔 and if that would be a case if that would cause some issue

Reply:
Maybe there's a slight possibility that a frame or two gets drawn before the root detection goes through.

But those frames should not contain any data yet as the ViewModel is still fetching data.

The alternative would be making the root detection completely synchronous. I believe that's unnecessary.

Reply:
I think we could revisit WireActivity anyways in the future and take this into account as I see some space for slight refactor